lib/system/basic_types.nim

```diff
@@ -1,18 +1,18 @@
 type
-  int* {.magic: Int.}         ## Default integer type; bitwidth depends on
+  int* {.magic: "Int".}         ## Default integer type; bitwidth depends on
```

Comment:
we don't need 2 ways of doing the same thing but maybe the non string way is better (because simpler) ?

Reply:
The string version doesn't universally work, for ``string* {.magic: "string".}`` the compiler used to complain no string type was available for "string".

Reply:
> The string version doesn't universally work

that doesn't seem incompatible with what i'm suggesting; Im suggesting:
* change all string version to identifier versions
* and optinally (like for importc), for the common case where identifier is predictable from symbol name (modulo capitalization of 1st letter), skip it

`type string* {.magic: String.}` is already in non-string mode; and, under optional rule, would become `type string* {.magic.}` since `"String"==capitalizeFirstLetter("string")`

```
proc `or`*(a, b: typedesc): typedesc {.magic: "TypeTrait", noSideEffect.}
```
would become (since optional shortcut doesn't apply):
```
proc `or`*(a, b: typedesc): typedesc {.magic: TypeTrait, noSideEffect.}
```